tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -50,6 +50,70 @@ where
         writer: &mut dyn fmt::Write,
         event: &Event<'_>,
     ) -> fmt::Result;
+
+    /// Write a log message for [creating a new span][new_span] with the given
+    /// `Attributes` in the given `Context` to the provided `Write`.
+    ///
+    /// This method is optional; the default impl is a no-op. Implementations
+    /// which wish to log calls to `new_span` can override this.
+    ///
+    /// [new_span]: https://docs.rs/tracing/0.1.9/tracing/subscriber/trait.Subscriber.html#tymethod.new_span
+    fn format_new_span(
+        &self,
+        _ctx: &FmtContext<'_, S, N>,
+        _writer: &mut dyn fmt::Write,
+        _attrs: &span::Attributes<'_>,
+    ) -> fmt::Result {
+        Ok(())
+    }
+
+    /// Write a log message for [closing a span][close] with the given
+    /// `Id` in the given `Context` to the provided `Write`.
+    ///
+    /// This method is optional; the default impl is a no-op. Implementations
+    /// which wish to log calls to `new_span` can override this.
+    ///
+    /// [close]: https://docs.rs/tracing/0.1.9/tracing/subscriber/trait.Subscriber.html#method.try_close
+    fn format_close(
+        &self,
+        _ctx: &FmtContext<'_, S, N>,
+        _writer: &mut dyn fmt::Write,
+        _id: &span::Id,
+    ) -> fmt::Result {
+        Ok(())
+    }
+
+    /// Write a log message for [entering a span][enter] with the given
+    /// `Id` in the given `Context` to the provided `Write`.
+    ///
+    /// This method is optional; the default impl is a no-op. Implementations
+    /// which wish to log calls to `new_span` can override this.
+    ///
+    /// [enter]: https://docs.rs/tracing/0.1.9/tracing/subscriber/trait.Subscriber.html#tymethod.enter
+    fn format_enter(
+        &self,
+        _ctx: &FmtContext<'_, S, N>,
+        _writer: &mut dyn fmt::Write,
+        _id: &span::Id,
+    ) -> fmt::Result {
+        Ok(())
+    }
+
+    /// Write a log message for [exiting a span][close] with the given
+    /// `Id` in the given `Context` to the provided `Write`.
+    ///
+    /// This method is optional; the default impl is a no-op. Implementations
+    /// which wish to log calls to `new_span` can override this.
+    ///
+    /// [close]: https://docs.rs/tracing/0.1.9/tracing/subscriber/trait.Subscriber.html#tymethod.exit
+    fn format_exit(
+        &self,
+        _ctx: &FmtContext<'_, S, N>,
+        _writer: &mut dyn fmt::Write,
+        _id: &span::Id,
+    ) -> fmt::Result {
+        Ok(())
+    }
```

Comment:
I think adding these to the trait is a much better approach than adding separate new traits for each lifecycle event. We may want to consider renaming the trait, as it now formats more than just the `Event` type...although, we can also think of "enter", "exit", "new_span", and "close" as "events" in the lowercase-E sense...

Note that if we decide to rename the trait that would also require a deprecation cycle.

Reply:
@hawkw do you mean that you would rather we deprecate `FormatEvent` in favor of a new trait that defines `format_event` similarly but also the newly added methods like `format_new_span`? If so, any suggestions for what this new trait would be called?:sweat_smile:

Reply:
> do you mean that you would rather we deprecate `FormatEvent` in favor of a new trait that defines `format_event` similarly but also the newly added methods like `format_new_span`?

Yes, that's what I meant, though I'm not sure whether or not it's what we _should_ do or not. Just a potential alternative to consider.